components/dialog/dialog-mixin.js

```diff
@@ -0,0 +1,85 @@
+export const DialogMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			opened: { type: Boolean, reflect: true },
+			title: { type: String },
+			width: { type: Number },
+			_state: { type: String, reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this.opened = false;
+		this._hasNativeDialog = (window.HTMLDialogElement !== undefined);
+		//this._hasNativeDialog = false;
+	}
+
+	attributeChangedCallback(name, oldval, newval) {
+		super.attributeChangedCallback(name, oldval, newval);
+		if (name === 'opened' && oldval !== newval) {
+			if (this.opened) this._open();
+			else this._close();
+		}
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this.addEventListener('keydown', this._handleKeyDown);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this.removeEventListener('keydown', this._handleKeyDown);
+	}
+
+	_close() {
```

Comment:
When `_close` gets called, we want to animate the dialog closing.  After the animation completes, we update the internal `_state`, public `opened`, and call native dialog's `close()` to actually hide the dialog in the DOM.  As per the native dialog spec, its `close()` must be called to properly close.